cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorTest.java

```diff
@@ -580,26 +580,26 @@ private DescribeLogDirsResult getDescribeLogDirsResult() {
       Constructor<DescribeLogDirsResult> constructor = DescribeLogDirsResult.class.getDeclaredConstructor(Map.class);
       constructor.setAccessible(true);
 
-      Map<Integer, KafkaFuture<Map<String, DescribeLogDirsResponse.LogDirInfo>>> futureByBroker = new HashMap<>();
-      Map<String, DescribeLogDirsResponse.LogDirInfo> logdirInfoBylogdir =  new HashMap<>();
-      Map<TopicPartition, DescribeLogDirsResponse.ReplicaInfo> replicaInfoByPartition = new HashMap<>();
-      replicaInfoByPartition.put(T0P0, new DescribeLogDirsResponse.ReplicaInfo(0, 0, false));
-      replicaInfoByPartition.put(T0P1, new DescribeLogDirsResponse.ReplicaInfo(0, 0, false));
-      replicaInfoByPartition.put(T1P0, new DescribeLogDirsResponse.ReplicaInfo(0, 0, false));
-      replicaInfoByPartition.put(T1P1, new DescribeLogDirsResponse.ReplicaInfo(0, 0, false));
-      logdirInfoBylogdir.put("/tmp/kafka-logs", new DescribeLogDirsResponse.LogDirInfo(Errors.NONE, replicaInfoByPartition));
+      Map<Integer, KafkaFuture<Map<String, LogDirDescription>>> futureByBroker = new HashMap<>();
+      Map<String, LogDirDescription> logdirInfoBylogdir =  new HashMap<>();
+      Map<TopicPartition, ReplicaInfo> replicaInfoByPartition = new HashMap<>();
+      replicaInfoByPartition.put(T0P0, new ReplicaInfo(0, 0, false));
+      replicaInfoByPartition.put(T0P1, new ReplicaInfo(0, 0, false));
+      replicaInfoByPartition.put(T1P0, new ReplicaInfo(0, 0, false));
+      replicaInfoByPartition.put(T1P1, new ReplicaInfo(0, 0, false));
+      logdirInfoBylogdir.put("/tmp/kafka-logs", new LogDirDescription(null, replicaInfoByPartition));
```

Comment:
This change would make this CC branch backwards incompatible wrt Kafka client version it uses -- i.e. if we would want to rollback the Kafka client version from `2.7` to `2.6`, the build would fail.

But we can avoid this by dropping the current reflection-based implementation in `getDescribeLogDirsResult()`. Instead, we can adopt a _mock_ based approach -- so rather than creating a `DescribeLogDirsResult`, can we use a mock of it and mock its responses in `#prepareContext(int, boolean)`, which seems to be the only place that this private method has been used?

Reply:
Thanks for the review. I just updated the PR where I fixed the issues.